### ISSUE TITLE
easy-rsa 3.0.9 Failed to install required data-files to PKI. (init)

### ISSUE THREAD

**Issue by chenrui333** [OpenVPN/easy-rsa#573]
Trying to upgrade to 3.0.9, but saw the `easyrsa init-pki` runtime failure:
```
==> easyrsa init-pki
* WARNING:

Failed to install required data-files to PKI. (init)


Easy-RSA error:

init-pki failed to create safe SSL conf: /private/tmp/easy-rsa-test-20220518-71956-ds9c54/pki/safessl-easyrsa.cnf
```

relates to https://github.com/Homebrew/homebrew-core/pull/101848

**Comment by TinCanTech** [comment]
Does the script have access to this data:
- `openssl-easyrsa.cnf`
- `x509-types` folder

**Comment by TinCanTech** [comment]
Please see #511  for further details.

**Comment by chenrui333** [comment]
what about `vars.example`, is that still in use. Somehow I still cannot get it working.

**Comment by ecrist** [comment]
On macOS simply extracting the tgz file, changing in to the easyrsa path and running the program works.

I'm not sure what you're trying to do in the brew build, but you're breaking it.

This isn't an easyrsa issue it's your build script in brew.

Eric Crist

On May 18, 2022, at 7:12 PM, Rui Chen ***@***.***> wrote:

﻿

what about vars.example, is that still in use. Somehow I still cannot get it working.

**Comment by TinCanTech** [comment]
> what about `vars.example`, is that still in use. Somehow I still cannot get it working.

If `vars.example` is missing that will still allow your PKI to be initialised.

`easyrsa` MUST find `openssleasyrsa.cnf` and `x509-types` folder.

Have them present in the same folder as `easyrsa` or one of these folders:
https://github.com/OpenVPN/easy-rsa/blob/cf32d9f4850d09152efb83d87f2230aa3b3ef83b/easyrsa3/easyrsa#L811-L818

**Comment by chenrui333** [comment]
Is `EASYRSA_PKI` configurable to find the PKI folder?

**Comment by chenrui333** [comment]
> On macOS simply extracting the tgz file, changing in to the easyrsa path and running the program works.
> I'm not sure what you're trying to do in the brew build, but you're breaking it.
> This isn't an easyrsa issue it's your build script in brew.

There are some brew env variables got setup during the build process to ensure the portability. Other than that I dont think we broke the build process, also this did not happen in 3.0.8. That is the reason why I raised an issue in here (more like a question, not as bug report)

**Comment by TinCanTech** [comment]
You can use `--pki-dir=DIR` or `set_var EASYRSA_PKI "foo"` in `vars`.

**Comment by SSgumS** [comment]
Hello
In `v3.0.8`, easyrsa automatically used the files existing beside the binary. But in `v3.0.9`, I must copy those data you mentioned (`openssl-easyrsa.cnf` and `x509-types`) to the working directory to get `init-pki` working.

**Comment by TinCanTech** [comment]
@SSgumS Please post the error message that you receive.

**Comment by SSgumS** [comment]
> @SSgumS Please post the error message that you receive.

The error message of command `easyrsa init-pki`:
```
* WARNING:

Failed to install required data-files to PKI. (init)


Easy-RSA error:

init-pki failed to create safe SSL conf: /app/pki/safessl-easyrsa.cnf

Host: nix | Linux |
```
What I did to resolve the error:
```
$ cp -r /usr/share/easyrsa/x509-types/ ./
$ cp /usr/share/easyrsa/openssl-easyrsa.cnf ./
```
Then the command is executed without any errors.

**Comment by TinCanTech** [comment]
> $ cp -r /usr/share/easyrsa/x509-types/ ./
> $ cp /usr/share/easyrsa/openssl-easyrsa.cnf ./

You have installed the files to the wrong folder.

The coorect name is n`easy-rsa`, with a dash `-`.

**Comment by TinCanTech** [comment]
For `v3.0.x` there could still be problems related to old packaging or other user error.  Such problems can be dealt with as they come to light.

**Comment by fireflycons** [comment]
I found that the way to resolve that is to run `easyrsa` in the directory in which it was installed.  Creating a soft link to it from e.g. `usr/local/sbin` doesn't work and that error is reported.

Surely the script can dereference the link if that is what is materialised as `$0` to get the real location and then use that?

**Comment by TinCanTech** [comment]
I'll take a look at v309

**Comment by TinCanTech** [comment]
@fireflycons I **cannot** reproduce any problem.

Can you explain exactly how you have installed Easy-RSA ?

**Comment by fireflycons** [comment]
I untarred it in `/opt` leaving the directory structure as per the tarball intact.
I exported env vars for `EASYRSA` pointing to `/opt/EasyRSA-3.1.0`,  and `EASYRSA_PKI` pointing to where I wanted PKI to go
Created a symlink `ln -s /opt/EasyRSA-3.1.0/easyrsa /usr/local/sbin/easyrsa`
Running from root's home: `easyrsa init-pki` and got the error as per the first message in this thread.

Then
```
cd /opt/EasyRSA-3.1.0
./easyrsa init-pki
```

This time no errors.
